modules/perception/camera/app/cipv_camera.h

```diff
@@ -0,0 +1,180 @@
+    const std::shared_ptr<base::Object> &object,
+    const EgoLane &egolane_ground,
+    LineSegment2Df *closted_object_edge);
+
+  // Check if the distance between lane and object are OK
+  bool AreDistancesSane(const float distance_start_point_to_right_lane,
+                        const float distance_start_point_to_left_lane,
+                        const float distance_end_point_to_right_lane,
+                        const float distance_end_point_to_left_lane);
+
+  // Check if the object is in the lane in image space
+  bool IsObjectInTheLaneImage(const std::shared_ptr<base::Object> &object,
+                              const EgoLane &egolane_image);
+  // Check if the object is in the lane in ego-ground space
+  //  |           |
+  //  | *------*  |
+  //  |         *-+-----*
+  //  |           |  *--------* <- closest edge of object
+  // *+------*    |
+  //  |           |
+  // l_lane     r_lane
+  bool IsObjectInTheLaneGround(const std::shared_ptr<base::Object> &object,
+                               const EgoLane &egolane_ground);
+
+  // Check if the object is in the lane in ego-ground space
+  bool IsObjectInTheLane(const std::shared_ptr<base::Object> &object,
+                         const EgoLane &egolane_image,
+                         const EgoLane &egolane_ground);
+
+  // Check if a point is left of a line segment
+  bool IsPointLeftOfLine(const Point2Df &point,
+                         const Point2Df &line_seg_start_point,
+                         const Point2Df &line_seg_end_point);
+
+  // Make a virtual lane line using a reference lane line and its offset
+  // distance
+  bool MakeVirtualLane(const LaneLineSimple &ref_lane_line,
+                       const float yaw_rate,
+                       const float offset_distance,
+                       LaneLineSimple *virtual_lane_line);
+
+  float VehicleDynamics(const uint32_t tick, const float yaw_rate,
+                        const float velocity, const float time_unit, float *x,
+                        float *y);
+  // Make a virtual lane line using a yaw_rate
+  bool MakeVirtualEgoLaneFromYawRate(const float yaw_rate,
+                                     const float velocity,
+                                     const float offset_distance,
+                                     LaneLineSimple *left_lane_line,
+                                     LaneLineSimple *right_lane_line);
+
+  // transform point to another using motion
+  bool TranformPoint(const Eigen::VectorXf& in,
+                     const Eigen::Matrix4f& motion_matrix,
+                     Eigen::Vector3d* out);
+
+  bool image2ground(const float image_x, const float image_y,
+    float *ground_x, float *ground_y);
+  bool ground2image(const float ground_x, const float ground_y,
+    float *image_x, float *image_y);
+
+  // Member variables
+  bool b_image_based_cipv_ = false;
+  int32_t debug_level_ = 0;
+  float time_unit_ = 0.0f;
+
+  const float EGO_CAR_WIDTH_METER = 1.86f;
```

Comment:
plase make this configuration configurable and globally shared

Reply:
Thank you for reviewing. I moved  variables in proto to be globally visible and configurable